### PR TITLE
Do not expose sensitive informations (json format)

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/api-platform/core/pull/978#issuecomment-290425327
| License       | MIT
| Doc PR        | na

https://github.com/api-platform/core/pull/978#issuecomment-290425327

### CODE REVIEW

File: src/Problem/Serializer/ErrorNormalizerTrait.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+namespace ApiPlatform\Core\Problem\Serializer;
+
+use Symfony\Component\Debug\Exception\FlattenException;
+use Symfony\Component\HttpFoundation\Response;
+
+trait ErrorNormalizerTrait
+{
+    private function getErrorMessage($object, bool $debug = false)
```
Comment (dunglas) [inline src/Problem/Serializer/ErrorNormalizerTrait.php]: `: string`?

**Comment by jordscream** [comment]
@dunglas @meyerbaptiste @soyuka , as I said in my [comment](https://github.com/api-platform/core/pull/978#issuecomment-288127293) in the previous PR (but nobody answered me ^^), does it work for a custom error normalizer  ? If I decide to use XML, it won't work ^^.

**Comment by soyuka** [comment]
`ErrorFormatGuesser` should fallback to the first available if none is registered no?

File: src/Problem/Serializer/ErrorNormalizerTrait.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+trait ErrorNormalizerTrait
+{
+    private function getErrorMessage($object, bool $debug = false): string
+    {
+        $message = $object->getMessage();
+
+        if ($debug) {
+            return $message;
+        } elseif ($object instanceof FlattenException) {
+            $statusCode = $context['statusCode'] ?? $object->getStatusCode();
```
Comment (meyerbaptiste) [inline src/Problem/Serializer/ErrorNormalizerTrait.php]: I think you forgot the `$context` var in the definition of your method 😉

Comment (soyuka) [inline src/Problem/Serializer/ErrorNormalizerTrait.php]: So we have no tests for context \o/ thanks @meyerbaptiste

**Comment by jordscream** [comment]
nobody to react to my comments ? :-(

**Comment by soyuka** [comment]
I answered @jordscream isn't this what the ErrorFormatGuesser is supposed to do?

File: src/Problem/Serializer/ErrorNormalizerTrait.php (hunk 2 above)
Comment (soyuka) [inline src/Problem/Serializer/ErrorNormalizerTrait.php]: resolved in last commit and I added a test

**Comment by dunglas** [comment]
Thank you very much @soyuka

**Pull request merged**